src/target/riscv/riscv-013.c

```diff
@@ -53,7 +53,8 @@ static int riscv013_write_debug_buffer(struct target *target, unsigned int index
 static riscv_insn_t riscv013_read_debug_buffer(struct target *target, unsigned int
 		index);
 static int riscv013_invalidate_cached_debug_buffer(struct target *target);
-static int riscv013_execute_debug_buffer(struct target *target);
+static int riscv013_execute_debug_buffer(struct target *target,
+		riscv_dbgbuf_exec_status_t *exec_status);
```

Comment:
I have not yet done proper review and will need to return to this later.

As an early feedback, please call this output value "execution result", not "execution status". This applies to all occurrences, not just this place.

I would also prefer to get the raw `cmderr` as the result, and not have a new enumeration for it. Same as you have it in `execute_abstract_command()`.

Reply:
> I would also prefer to get the raw cmderr as the result, and not have a new enumeration for it. Same as you have it in execute_abstract_command().

Unless there is a strong pushback - I would prefer to leave this part as is. The reason is that it renders existing API inconsistent.

Reply:
Which API do you refer to, please? Do you mean the `write_debug_buffer` function pointer in `struct riscv_info`?

If so, then I wouldn't worry about passing the cmderr, for these reasons:
- There is no debug buffer in 0.11 target support at all.
- This API based on function pointers to 0.11 and 0.13 functions I would like to eventually remove and rewrite by direct function calls since the current one makes the code reading (and code analysis by static tools) difficult and having this API brings no benefit.

Reply:
> Which API do you refer to, please? Do you mean the write_debug_buffer function pointer in struct riscv_info?

No. I believe I've misunderstood your original comment. I though that you've proposed to return cmderr instead of a generic `int` error code. Looks like  you just don't want `cmderr` to be translated to `riscv_dbgbuf_exec_status_t`. I haven't though about this possibility - will take a look.

Reply:
>  Looks like you just don't want cmderr to be translated to riscv_dbgbuf_exec_status_t

Yes, that was my idea.

I believe the low-level program buffer execution function should provide the raw error code (cmderr). If the caller wishes to translate the error code, that is OK, but the raw value should remain accessible.

Reply:
addressed * . There is a caveat, though:

1. low-level `execute_debug_buffer` returns raw `abstractcs_err`.
2. ` riscv_program_exec` still "returns" translated `abstractcs_err`. The reason is that it can fail during the preparation stage, while the respected abstract command is not ready to be executed. This corresponds to ` RISCV_DBGBUF_EXEC_RESULT_NOT_EXECUTED` enumeration value. I believe that it would be beneficial to clearly identify at what stage  `riscv_program_exec` fail.